tests/rptest/tests/describe_topics_test.py

```diff
@@ -167,38 +148,65 @@ def test_describe_topics_with_documentation_and_types(self):
             )
         }
 
-        for line in property_lines:
-            col = line.strip().lower().split()
-            self.logger.debug(col)
-            assert len(col) >= 4
-            name = col[0]
-            config_type = col[1]
-            value = col[2]
-            source_type = col[3]
-
-            # The above .split() also applies to the value for datapolicy
-            # so we need to re-concatenate the string
-            if name == "redpanda.datapolicy":
-                assert len(col) == 5
-                value = f'{col[2]} {col[3]}'
-                source_type = col[4]
-
-            assert name in properties.keys()
-            prop = properties[name]
-            assert config_type == prop.config_type
-            assert value == prop.value
-            assert source_type == prop.source_type
-
-        # The output format follows:
+        tp_spec = TopicSpec()
+        self.client().create_topic([tp_spec])
+        kcl = KCL(self.redpanda)
+        output = kcl.describe_topic(tp_spec.name,
+                                    with_docs=True,
+                                    with_types=True).lower().splitlines()
+        self.logger.debug(f"Raw output\n{output}")
+
+        # First thirteen lines follow a TSV-like format and contain the property names,
+        # config types, source, and value
+        property_lines = output[:13]
+        # Lines after that follow a different format and have the documentation strings
+        documentation_lines = output[13:]
+        self.logger.debug(f"Properties\n{property_lines}")
+        self.logger.debug(f"Documentation\n{documentation_lines}")
+
+        self.logger.debug("Parsing documentation")
+        # The documentation format follows:
         # line 1 is empty
         # line 2 is the property name
         # line 3 is the property docstring
         prop = None
-        for i in range(0, len(documentation_lines), 3):
+        for line in documentation_lines:
```

Comment:
It took me a while to understand what this loop was doing. It sets some mutable state to be ready in the next iteration (prop). I personally think it would be cleaner for the internal loop logic to be identical across all iterations with respect to reaching out and setting/getting mutable state.

I'd consider taking a more functional programming approach. I would zip() together the list into logical groups, most likely 2-tuples. Then perform the validator on those logical chunks:
```
# Step over list every 3 items
zipped_lines = [(doc_lines[i+1], doc_lines[i+2]) for lines in range(0, len(doc_lines, 3)]
for prop_line, doc_line in zipped_lines:
    property_name = grab_property(prop_line)
    assert property_name in properties
    assert doc_line == properties[property_name]
```

Reply:
imho its even more explicit with

```
assert len(doc_lines)%3==0
for i in range(0, len(doc_lines)/3):
  property = doc_lines[3*i]
  description = doc_lines[3*i+1]
  empty_line = doc_lines[3*i+2]
  # asserts on property, description & empty_line
```